protocol/notifications.proto

```diff
@@ -0,0 +1,320 @@
+syntax = "proto3";
+package river;
+option go_package = "github.com/river-build/river/core/node/protocol";
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+
+service AuthenticationService {
+  rpc StartAuthentication(StartAuthenticationRequest) returns (StartAuthenticationResponse);
+
+  rpc FinishAuthentication(FinishAuthenticationRequest) returns (FinishAuthenticationResponse);
+}
+
+message StartAuthenticationRequest {
+  // user id of the user that wants to authenticate
+  bytes user_id = 1;
+}
+
+message StartAuthenticationResponse {
+  // user id of the user that wants to authenticate
+  bytes user_id = 1;
+  // challenge that must be signed.
+  // clients must verify before signing that the challenge is >= 128 bits
+  bytes challenge = 2;
+  // indication until when challenge is valid
+  google.protobuf.Timestamp expiration = 3;
+}
+
+// note, challenge expiration is stored in the notification service and therefore doesn't have to be provided
+// in the request. Expiration is looked up through the nonce.
```

Comment:
you're using both the word nonce and the word challenge to mean the same thing. Can we call it challenge everywhere?